### ISSUE TITLE
Heatmap colorbar should respect value of 'ticks'  in cbar_kws

### ISSUE THREAD

**Issue by hanslovsky** [mwaskom/seaborn#459]
When you add a `'ticks'` keyword to `cbar_kws`, python will crash because the keyword `'ticks'` appears twice in the same funciton call. In order to avoid that, default ticks should only be created, if `'ticks'` is not present in `cbar_kws`.

```python
import seaborn as sns
import numpy as np
from matplotlib import pyplot as plt

mat = np.random.randn( 20, 20 )
cbar_kws = { 'ticks' : [ np.min(mat), np.max(mat) ] }
sns.heatmap( mat, square=True, cbar_kws=cbar_kws )
plt.show()
```
This code snippet fail with
```bash
/usr/lib/python2.7/site-packages/matplotlib/__init__.py:479: UserWarning: matplotlibrc text.usetex option can not be used unless TeX-3.1415 or later is installed on your system
  'installed on your system') % tex_req)
/usr/lib/python2.7/site-packages/matplotlib/__init__.py:485: UserWarning: matplotlibrc text.usetex can not be used with *Agg backend unless dvipng-1.5 or later is installed on your system
  warnings.warn( 'matplotlibrc text.usetex can not be used with *Agg '
/usr/lib/python2.7/site-packages/matplotlib/__init__.py:499: UserWarning: matplotlibrc text.usetex can not be used unless ghostscript-7.07 or later is installed on your system
  'installed on your system') % gs_req)
Traceback (most recent call last):
  File "seaborn-example.py", line 9, in <module>
    sns.heatmap( mat, square=True, cbar_kws=cbar_kws )
  File "/home/phil/local/lib/python2.7/site-packages/seaborn-0.6.dev0-py2.7.egg/seaborn/matrix.py", line 292, in heatmap
    plotter.plot(ax, cbar_ax, kwargs)
  File "/home/phil/local/lib/python2.7/site-packages/seaborn-0.6.dev0-py2.7.egg/seaborn/matrix.py", line 193, in plot
    ticks=ticker, **self.cbar_kws)
TypeError: colorbar() got multiple values for keyword argument 'ticks'
```

Compare this commit that solved the problem:
5761bf0c7a9ad6e6b0b67209f83d160228905d16

If you agree on that issue, I'll create a pull request.

**Comment by mwaskom** [comment]
Sure, that makes sense. A more elegant way to do the same thing would also be:

```python
default_ticker = mpl.ticker.MaxNLocator(6)
self.cbar_kws.setdefault('ticks', default_ticker)
```

Also a test would be good.

**Comment by hanslovsky** [comment]
Sounds good, will use the more pythonic code and add a test later.

**Comment by mwaskom** [comment]
Any progress here?

**Comment by hanslovsky** [comment]
Got distracted by work, but I'll add it.

**Comment by jackgolding** [comment]
Are there any immediate work arounds for this?

**Comment by hanslovsky** [comment]
I finally got the pull-request done. @jackgolding the changes are available in https://github.com/hanslovsky/seaborn at the moment and on this repo after the pull request has been merged.

**Comment by mwaskom** [comment]
Closed with #869
